sycl/source/detail/scheduler/commands.cpp

```diff
@@ -1812,9 +1839,27 @@ cl_int ExecCGCommand::enqueueImp() {
     for (ArgDesc &Arg : ExecKernel->MArgs) {
       switch (Arg.MType) {
       case kernel_param_kind_t::kind_accessor: {
+// TODO ESIMD Currently all accessors are treated as ESIMD under corresponding
+// compiler option enabling the macro below. Eventually ESIMD kernels and usual
+// kernels must co-exist and there must be a mechanism for distinguishing usual
+// and ESIMD accessors.
+#ifndef __SYCL_EXPLICIT_SIMD__
+        constexpr bool IsESIMDAcc = false;
+#else
+        constexpr bool IsESIMDAcc = true;
+#endif // __SYCL_EXPLICIT_SIMD__
         Requirement *Req = (Requirement *)(Arg.MPtr);
         AllocaCommandBase *AllocaCmd = getAllocaForReq(Req);
         RT::PiMem MemArg = (RT::PiMem)AllocaCmd->getMemAllocation();
+
+        if (IsESIMDAcc) {
+          // Create additional wrapper object for an ESIMD accessor and record
+          // it for subsequent de-allocation.
+          size_t AllocSize = Req->MMemoryRange.size() * Req->MElemSize;
+          MemArg = createImage1dBufferWrapper(*getSyclObjImpl(Context),
```

Comment:
Please, make sure that such combined alloca(buffer + image) works correctly in cases when we move memory between contexts.

Reply:
As agreed, I will add a TODO - ESIMD interop will need to be addressed.

Reply:
@s-kanaev, please review. I think I addressed all @romanovvlad's comments, except the dummy opencl_image1d_buffer_type specialization for atomic mode, which I can't find simple solution for at this point, and added a TODO.